### ISSUE TITLE
Astro@1.6.3 doesn't redirect localhost:3000/ as an index file when using [...slug].astro files

### ISSUE THREAD

**Issue by hayhaydz** [withastro/astro#5300]
### What version of `astro` are you using?

1.6.3

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

I've brutally copied the title of a closed issue [#1187](https://github.com/withastro/astro/issues/1187) because it seems as though it fits that issues description. The problem is that when Astro builds a `[...slug].astro` file with a slug of `index` that is stored under the pages directory it will build as so `/index/index.html` instead of the expected outcome of `/index.html`, a temporary solution is to use this in the build configuration:

```
build: {
  format: 'file'
},
```

For my particular case I'm working with a CMS and am retrieving all of the available pages slugs and ids but I also took the above mentioned closed issue's recreated CodeSandbox and used the latest version of Astro and could produce the same unexpected result.

### Link to Minimal Reproducible Example

https://codesandbox.io/s/crazy-bouman-2fi6eu?file=/src/pages/%5B...id%5D.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
I would kind of expect `param: ''` to work here. I would never expect `'index'` to be interpreted as a non-subfolder route. https://stackblitz.com/edit/github-9xqq8p?file=src%2Fpages%2F[...id].astro&on=stackblitz

**Comment by wulinsheng123** [comment]
> I would kind of expect `param: ''` to work here. I would never expect `'index'` to be interpreted as a non-subfolder route. [https://stackblitz.com/edit/github-9xqq8p?file=src%2Fpages%2F[...id].astro&on=stackblitz](https://stackblitz.com/edit/github-9xqq8p?file=src%2Fpages%2F%5B...id%5D.astro&on=stackblitz)

I have little confused for expecting ‘param:’’’ could you tell me more if you don’t mind

**Comment by wulinsheng123** [comment]
> I would kind of expect `param: ''` to work here. I would never expect `'index'` to be interpreted as a non-subfolder route. [https://stackblitz.com/edit/github-9xqq8p?file=src%2Fpages%2F[...id].astro&on=stackblitz](https://stackblitz.com/edit/github-9xqq8p?file=src%2Fpages%2F%5B...id%5D.astro&on=stackblitz)

I want to confirm this question with you. So If I set another `id`, We are interpreted as a non-subfolder router? right?

**Comment by MoustaphaDev** [comment]
@matthewp I don't think this problematic though since we can already do that by passing in `undefined`, it's a nice to have IMO.

**Comment by wulinsheng123** [comment]
> @matthewp I don't think this problematic though since we can already do that by passing in `undefined`, it's a nice to have IMO, we can just document that.

What’s IMO?

**Comment by MoustaphaDev** [comment]
> > @matthewp I don't think this problematic though since we can already do that by passing in `undefined`, it's a nice to have IMO, we can just document that.
> 
> What’s IMO?

Sorry, edited!

**Comment by matthewp** [comment]
Ah, yeah so if `undefined` does it then we should document that case. Going to transfer this issue over to docs.
